ros_buildfarm/common.py

```diff
@@ -575,9 +576,20 @@ def get_packages_in_workspaces(workspace_roots, condition_context):
 
 
 def get_package_repo_data(repository_baseurl, targets, cache_dir):
+    get_index_methods = {
+        'deb': get_debian_repo_index,
+        'rpm': get_rpm_repo_index,
+    }
+
     data = {}
     for target in targets:
-        index = get_debian_repo_index(
+        package_format = package_format_mapping.get(target.os_name)
+        get_index_method = get_index_methods.get(package_format)
+        if not get_index_method:
+            raise RuntimeError(
```

Comment:
I feel like I'm going to end up with a lot of checks like this. Would you rather I just index directly into the maps and allow the `KeyError` to bubble up if something is missing?

Reply:
Raising a `KeyError` in this case (which should never happen) sounds fine to me.

Reply:
OK, updated to remove the extra logic.